### DISCUSSION TITLE
Prompt for calling the method of connecting to the database

### DISCUSSION THREAD

**Discussion by bouyeijiang** [actix/actix-web#2826] [Q&A]
actix-rt|system:0|arbiter:0' panicked at 'Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks

**Comment by robjtede** [comment]
There's not a lot we can do to help without knowing a lot more about your code. Either way, you can't call `block_on` within a request handler so if you are doing that somewhere it needs to be replaced with an async version. Eg. using `reqwest::blocking` is a common mistake like this.

**Comment by bouyeijiang** [comment]
the code method is shown in the following picture
![a](https://user-images.githubusercontent.com/4083760/181502908-e1166e69-25e5-42c5-8f43-6c7148c78ced.png)

**Comment by bouyeijiang** [comment]
query database method:

```rust
pub fn db_query_r(conf: &mut Dbcfg, command_text: &str) -> Vec<Row> {
    let mut client = db_connect(conf).unwrap();

    let rt = db_query(&mut client, command_text).unwrap();

    rt
}
```

**Reply by robjtede** [reply]
Almost certainly that `db_connect` is causing the problem. What crate are you using for accessing the database? Are `db_connect` and `db_query` your functions?

**Comment by bouyeijiang** [comment]
postgres="0.19.2"

yeah!

```rust
pub fn db_query(client: &mut Client, command_text: &str) -> Result<Vec<Row>, Error> {
    let rt = client.query(command_text, &[]);
    rt
}
```

**Reply by robjtede** [reply]
> This crate is a lightweight wrapper over tokio-postgres. The postgres::Client is simply a wrapper around a tokio_postgres::Client along side a tokio Runtime. The client simply blocks on the futures provided by the async client.

- https://docs.rs/postgres/latest/postgres/#implementation

So you need to use [`tokio-postgres`](https://docs.rs/tokio-postgres/latest/tokio_postgres/) instead.

**Comment by bouyeijiang** [comment]
async/await, i see. thinks!
